beacon_node/network/src/beacon_processor/worker/rpc_methods.rs

```diff
@@ -247,35 +247,38 @@ impl<T: BeaconChainTypes> Worker<T> {
                             );
                         }
                         Ok((Some(block), None)) => {
-                            let data_availability_boundary_by_root = self.chain.finalized_data_availability_boundary();
+                            let finalized_data_availability_boundary = self.chain.finalized_data_availability_boundary();
                             let block_epoch = block.epoch();
 
-                           if Some(block_epoch) >= data_availability_boundary_by_root {
+                           if Some(block_epoch) >= finalized_data_availability_boundary {
                                 debug!(
                                     self.log,
                                     "Peer requested block and blob that should be available, but no blob found";
                                     "peer" => %peer_id,
                                     "request_root" => ?root,
-                                    "data_availability_boundary_by_root" => data_availability_boundary_by_root,
+                                    "finalized_data_availability_boundary" => finalized_data_availability_boundary,
                                 );
                                 self.send_error_response(
                                     peer_id,
                                     RPCResponseErrorCode::ResourceUnavailable,
-                                    "No blob for requested block.".into(),
+                                    "Blobs unavailable".into(),
                                     request_id,
                                 );
+                               send_response = false;
+                               break;
                             } else {
                                 debug!(
                                     self.log,
-                                    "Peer requested block and blob older than the data availability boundary for ByRoot request, no blob found";
+                                    "Peer requested block and blob older than the data availability \
+                                    boundary for ByRoot request, no blob found";
                                     "peer" => %peer_id,
                                     "request_root" => ?root,
-                                    "data_availability_boundary_by_root" => data_availability_boundary_by_root,
+                                    "finalized_data_availability_boundary" => finalized_data_availability_boundary,
                                 );
                                 self.send_error_response(
                                     peer_id,
                                     RPCResponseErrorCode::ResourceUnavailable,
-                                    format!("No blob for requested block. Requested blob is older than the data availability boundary for a ByRoot request, currently at epoch {:?}", data_availability_boundary_by_root),
```

Comment:
I truncated this because the max error length looks like it's 32 bytes

Reply:
How did you realise it's longer than 32 bytes?

Reply:
I think UTF-8 is usually about one byte per character so I made a guess here :slightly_smiling_face:

Reply:
Oh right, human eye - dangerous. I will keep that in mind. Would be nice with a lint to check these error messages...